src/pj_gc_reader.c

```diff
@@ -199,7 +212,7 @@ static int pj_gc_readentry(projCtx ctx, PAFile fid, PJ_GridCatalogEntry *entry)
     }
```

Comment:
After the `token_count = pj_gc_read_csv_line(...)`, and given your focus on memory handling in this PR, you probably should loop over all tokens to check for `0`, since `pj_gc_read_csv_line` does not check whether its calls to `strdup()` succeeds.

Reply:
In `pj_gc_read_csv_line` I added code to stop if `strdup` ever fails. However, we should probably make sure that error is propagated properly.

However, no more work should be necessary here, all `tokens` from `0` to `token_count - 1` should be non-null.

Reply:
Yes - that's a better solution!